lib/ua-parser.ts

```diff
@@ -110,7 +111,12 @@ const parseUA = (userAgent: string, browsers: Browsers): ParsedUserAgent => {
   }
 
   data.fullVersion = data.fullVersion || ua.browser.version || "0";
-  data.version = getMajorMinorVersion(data.fullVersion);
+
+  if (RUNTIME_IDS_WITH_PATCH_VERSIONING.has(data.browser.id)) {
+    data.version = data.fullVersion;
+  } else {
+    data.version = getMajorMinorVersion(data.fullVersion);
+  }
 
   if (data.browser.id == "bun") {
```

Comment:
This line and the following two lines were my quick hack to do something similar, but I like your solution much better as it doesn't special-case Bun!